tests/llm/test_nous_refresh.py

```
@@ -0,0 +1,159 @@
+    monkeypatch.setitem(sys.modules, "src.config", SimpleNamespace(settings=fake_settings))
+
+    try:
+        result = await refresh_nous_credentials()
+    finally:
+        # Restore
+        refresh_mod.save_state = original_save
+        refresh_mod.update_env_key = original_update_env
+        refresh_mod._find_project_root = original_find_root
+
+    assert result == "new_agent_key_123"
+    assert saved_state["refresh_token"] == "new_refresh_token"
```

Ruff (0.15.14)

[error] Possible hardcoded password assigned to: "refresh_token"

(S105)